core/string.cpp

```diff
@@ -45,4 +53,35 @@ std::map<DType, std::string> &dtypeNameRegistry() {
   return registry;
 }
 
+const std::string to_iso_date(const scipp::core::time_point &item,
+                              const std::optional<units::Unit> &unit) {
+  if (!unit)
+    throw except::UnitError(
+        "Time point should only have time units (ns or s).");
+
+  int64_t ts = item.time_since_epoch();
+  if (unit.value() == units::ns) {
+    int64_t conv = 1000000000;
+    // time representation of the timestamp
+    int64_t time = ts / conv;
+    auto tm = *std::gmtime(&time);
```

Comment:
Not sure what the OSX issue is, but maybe this can be rewritten with utilities from `<chrono>` instead of `<ctime>`?